pubsublite/ps/integration_test.go

```diff
@@ -0,0 +1,673 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package ps
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/internal/testutil"
+	"cloud.google.com/go/internal/uid"
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/pubsublite"
+	"cloud.google.com/go/pubsublite/internal/test"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"google.golang.org/api/option"
+
+	vkit "cloud.google.com/go/pubsublite/apiv1"
+	pb "google.golang.org/genproto/googleapis/cloud/pubsublite/v1"
+)
+
+const (
+	gibi               = 1 << 30
+	defaultTestTimeout = 5 * time.Minute
+)
+
+var (
+	resourceIDs = uid.NewSpace("go-ps-test", nil)
+	rng         *rand.Rand
+
+	// A random zone is selected for each integration test run.
+	supportedZones = []string{
```

Comment:
Since we just turned up a few new zones, I wonder if you could include some of those here to get more coverage. I don't think we'll keep this in sync, but I'd like to have more confidence that the new zones are working well.

Reply:
I ran the integration tests on each of the new zone and they all worked perfectly!

Good idea, added the new zones and also moved the list to test utils to share with the admin integration tests. Having more zones will reduce the likelihood of quota issues, if many integration tests happen to be running simultaneously.